corehq/apps/domain/tests/test_password_strength.py

```diff
@@ -4,25 +4,30 @@
 from corehq.apps.domain.forms import clean_password
 
 
+@override_settings(MINIMUM_PASSWORD_LENGTH=0, MINIMUM_ZXCVBN_SCORE=2)
 class PasswordStrengthTest(SimpleTestCase):
 
-    @override_settings(MINIMUM_ZXCVBN_SCORE=2)
     def test_score_0_password(self):
         self.assert_bad_password(PASSWORDS_BY_STRENGTH[0])
 
-    @override_settings(MINIMUM_ZXCVBN_SCORE=2)
     def test_score_1_password(self):
         self.assert_bad_password(PASSWORDS_BY_STRENGTH[1])
 
-    @override_settings(MINIMUM_ZXCVBN_SCORE=2)
     def test_score_2_password(self):
         self.assert_good_password(PASSWORDS_BY_STRENGTH[2])
 
-    @override_settings(MINIMUM_ZXCVBN_SCORE=2)
+    @override_settings(MINIMUM_PASSWORD_LENGTH=8)
+    def test_length_2_password(self):
+        self.assert_bad_password(PASSWORDS_BY_STRENGTH[2])
+
+    @override_settings(MINIMUM_PASSWORD_LENGTH=8)
     def test_score_3_password(self):
         self.assert_good_password(PASSWORDS_BY_STRENGTH[3])
 
-    @override_settings(MINIMUM_ZXCVBN_SCORE=2)
+    @override_settings(MINIMUM_PASSWORD_LENGTH=12)
+    def test_length_3_password(self):
+        self.assert_bad_password(PASSWORDS_BY_STRENGTH[3])
+
```

Comment:
Hi Matt, thank you a lot for pointing this out. I didn't get the original naming of the test at first. After read your comments, at the bottom of the original testing of "password score", I added 1 test for password length, and 3 test respectively for your 3 scenarios.

> 0 seems like a special condition where the check should be disabled entirely, but I don't know that that test would look any different from a password that meets the minimum length requirement

Are you referring to `override_settings(MINIMUM_PASSWORD_LENGTH=0)` ? Because I want to let the original test still test the zxcvbn library only, so I disabled check on minimum password length.

Reply:
> `test_length_2_password` sounds to me like we're testing whether a password with 2 characters will be considered a good password or not, but that doesn't seem to be what's occurring here

As best as I can tell, the tests were originally named by the _index into `PASSWORDS_BY_STRENGTH` of the password being tested_. Jing and I talked about this when she first set out to write these tests and I wasn't a fan of that naming convention, but recommended to "leave it as-is" for the sake of code review (i.e. don't rename a bunch of existing tests when you haven't changed what they're testing).

My recommendation here would be to name all the _new_ tests according to what they _do_ (don't use the index of the list the password is coming from). This would make the tests much clearer to me.